### PR TITLE
Bump inversify

### PR DESCRIPTION
**Pull request by joyceerhl**
Related to https://github.com/microsoft/vscode-jupyter/issues/4144

The version we're on is 2 years old and has been deprecated.

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1814,7 +1814,7 @@
         "glob": "^7.1.2",
         "hash.js": "^1.1.7",
         "iconv-lite": "^0.4.21",
-        "inversify": "^4.11.1",
+        "inversify": "^5.0.1",
```
Comment (IanMatthewHuff) [inline package.json:1817]: No reason not to approve this. But I looked it up out of interested. Looks like this version is also basically two years old.  Looks like there was an update to the inversify js base this summer that has not made it to the npm package yet.

Comment (IanMatthewHuff) [reply package.json:1817]: From the commits doesn't look like much really changed. So probably not missing on much there.

**Review by IanMatthewHuff — APPROVED** [review]

**Comment by rchiodo** [comment]
Strange that the code coverage went up by switching inversify?

**Review by DavidKutu — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (joyceerhl) [reply package.json:1817]: Yeah, I'm mainly interested in this change:
https://github.com/inversify/InversifyJS/blob/master/CHANGELOG.md#changed

> Change GUID to incremented counter for better performance #882

**Comment by joyceerhl** [comment]
@rchiodo I've seen the code coverage change with small commits (on other PRs) so not totally sure what's going on there.

**Pull request merged**